USING_WITH_PYRIGHT.md

```diff
@@ -0,0 +1,305 @@
+
+## Settings differences
+
+Both Pylance and Pyright have [settings](https://github.com/microsoft/pylance-release#settings-and-customization) to control behavior. For the most part, they use the same values, but the few differences do have an impact on the analysis.
+
+| Setting | Pylance default | Pyright default | Description | Potential Impact |
+|----|----|----|----|----|
+| python.analysis.autoSearchPaths | true | false|  Adds 'src' to the list of search paths | This may change what files are found when analyzing. So if you're getting missing imports for things in your 'src' tree, this might be why |
+| python.analysis.indexing | true | false | All files in the current virtual environment are parsed to a certain depth. This gives Pylance more information for auto import completions and other things. | This can have an impact on completions and hover information, but shouldn't impact errors returned.
+| python.analysis.useLibraryCodeForTypes | true | false | When true, this means 3rd party libraries are analyzed to produce type information. Without it, all 3 party libraries are just assumed to be of type Any unless they provided type information explicitly | This setting changes the types found by Pylance/Pyright. It can definitely cause differences in errors reported, so make this the same for both if you want consistent results. A lot of the time this is the sole cause of differences. |
+| python.analysis.typeCheckingMode | off | basic | Determines what diagnostics are shown. | Pylance defaults to off, but there is a possibility that VS code will force this to 'basic' for pylance users. If you want to guarantee this is the same as Pyright, force it to 'basic' (or whatever you want to enforce) by specifying it in your settings.json|
+| python.analysis.diagnosticMode | openFilesOnly | openFilesOnly | Determines which files are analyzed. `openFilesOnly` means only those files that are open in an IDE | You'll likely set this to 'workspace' for the CI version of Pyright. This means Pylance would likely not give you all of the same errors since it's only looking at open files. |
+| python.analysis.diagnosticSeverityOverrides | - | - | Allows a user to override the severity levels for individual diagnostic rules. | See below |
+
+## Diagnostic severity overrides
+
+Both Pylance and Pyright specify a custom [`python.analysis.diagnosticSeverityOverrides`](https://microsoft.github.io/pyright/#/configuration?id=diagnostic-rule-defaults) setting. Differences in this setting have a direct impact on the diagnostics returned. If you were attempting to get the same error output for both, making sure this setting is consistent is probably the first thing to do.
+
+
+### Diagnostic Severity Overrides to match Pyright defaults
+
+Here's the settings you would provide to Pylance based on which Pyright mode you are using:
+
+<details>
+  <summary>Pyright 'off' mode</summary>
+
+```json
+{
+    "python.anaylsis.diagnosticSeverityOverrides": {

```

Comment:
Could we limit this to just the diagnostics which differ? I thought `reportShadowedImports` was the only one that was different? But maybe I'm misunderstanding what issue this is addressing.

Reply:
No a bunch of them differ. And it was easier to just list this out than try and keep track of the differences.

Reply:
Oh, I forgot about `reportShadowedImports`. That one does differ between pyright and pylance, doesn't it? 

If so, we should consider removing this difference. If we decide to leave the difference, then I agree with Erik that we should simply document the difference rather than repeating all of the default values that do not differ.

Reply:
Weird. Based on these comments from Eric I thought it was just that one. Maybe we should consider getting these back in sync?

https://github.com/microsoft/pylance-release/issues/3857#issuecomment-1407167773
https://github.com/microsoft/pylance-release/issues/3857#issuecomment-1407208449

Reply:
Maybe this was the wrong spot to look, but if you compare the default values in https://github.com/microsoft/pyright/blob/main/packages/vscode-pyright/package.json with the values in https://github.com/microsoft/pyrx/blob/main/packages/vscode-pylance/package.json, they differ in like 12 spots

Reply:
Hmm, that's the documentation, not the source of truth. I think if you look at the source of truth in the code (https://github.com/microsoft/pyright/blob/09c79ed3e284b953b4d96c240a2dca91d74b2eee/packages/pyright-internal/src/common/configOptions.ts#L421), I think (hope?) that you'll see more consistency. If you're finding many differences in the documentation, we should fix those as well to ensure that they're consistent with the behaviors defined in the code.

Reply:
Okay I'll enter a bug to fix the documentation in the pylance extension. Althought it's kind of ambiguous there too because the default type checking mode is off, unless you're in an experiment, then it's basic. Which changes the default value for a bunch of settings.

Reply:
I thought that experiment ended a long time ago and now everyone was opted in to "basic" by default. Is that not the case? If not, why is the experiment still going after more than a year? What additional information is needed to make a decision?

Reply:
Not sure myself, @luabud do you know if this is at 100% or not? Perhaps we can just eliminate the experiment now?